cargo-marker/src/cli.rs

```diff
@@ -135,27 +93,28 @@ mod tests {
     }
 
     #[test]
-    fn test_split_args() {
-        let (a, b) = split_args(vec![]);
-        assert_eq!(a, &[] as &[&str]);
-        assert_eq!(b, &[] as &[&str]);
-
-        // The `marker` argument is an artifact from the invocation from cargo.
-        // It should be removed, if it's on the second position
-        let (a, b) = split_args(["cargo", "marker", "check"].iter().map(ToString::to_string));
-        assert_eq!(a, &["cargo", "check"] as &[&str]);
-        assert_eq!(b, &[] as &[&str]);
-
-        let (a, b) = split_args(["check", "--something", "marker"].iter().map(ToString::to_string));
-        assert_eq!(a, &["check", "--something", "marker"] as &[&str]);
-        assert_eq!(b, &[] as &[&str]);
-
-        let (a, b) = split_args(["check", "--", "cargo", "magic"].iter().map(ToString::to_string));
-        assert_eq!(a, &["check"] as &[&str]);
-        assert_eq!(b, &["cargo", "magic"] as &[&str]);
-
-        let (a, b) = split_args(["check", "--"].iter().map(ToString::to_string));
-        assert_eq!(a, &["check"] as &[&str]);
-        assert_eq!(b, &[] as &[&str]);
+    fn test_marker_cli() {
+        let cli = MarkerCli::parse_from(&["cargo-marker", "check"]);
+        assert!(matches!(cli.command, Some(CliCommand::Check(_))));
+
+        let cli = MarkerCli::parse_from(&["cargo-marker"]);
+        assert!(matches!(cli.command, None));
```

Comment:
The `assert_maches` link points to the CLI code of next test and I can't find the create on crates.io

Something like this would definitely be useful. You already mentioned this in a previous PR and https://github.com/rust-marker/marker/issues/202 exists to investigate it. I would rather not add it in this PR, to keep it more contained, but this would be a great application for it :+1: (I just need time to get through all the issues :sweat_smile: )

Reply:
Yeah, an apology for the typo. It's [`assert_matches`](https://docs.rs/assert_matches/latest/assert_matches/)

Reply:
I think [expect_test](https://docs.rs/expect-test/latest/expect_test/) would be a better fit here, mostly because it allows automatic test updates. Still, I won't add it in this PR. My next weeks will be more busy, and I want to make sure this and the other stuff is added before 0.2.0.